source/NPC.cpp

```diff
@@ -673,6 +793,40 @@ NPC NPC::Instantiate(const PlayerInfo &player, map<string, string> &subs, const
 		result.system = location.PickSystem(origin);
 	if(!result.system)
 		result.system = (isAtDestination && destination) ? destination : origin;
+
+	if(missingWaypoint)
+		result.waypoints.push_back(result.destination);
+	for(const LocationFilter &filter : waypointFilters)
+	{
+		// Each new waypoint calculates "distance X Y" from the previous waypoint / origin.
```

Comment:
Comment just repeats what the code already says. And, I don't think it describes it very well, anyway.
```suggestion
```
If the code is really that difficult to understand:
```suggestion
		// Each waypoint is used as the center for the filter that selects the next waypoint;
		// the origin of the mission is used as the center for selecting the first waypoint.
```
Similar for the comments on stopovers and the destination.